### PR TITLE
ENTESB-10894, ENTESB-10896

### PR DESCRIPTION
**Pull request by rareddy**

### CODE REVIEW

**Comment by rareddy** [comment]
BTW, this does not affect any of UI code.

File: server/komodo-server/src/main/java/org/komodo/openshift/TeiidOpenShiftClient.java [hunk 1]
```diff
@@ -586,10 +581,8 @@ public void createDataSource(DefaultSyndesisDataSource scd) throws Exception {
         }
 
         //now that the komodoname is set, we can create the properties
-        Map<String, String> properties = scd.convertToDataSourceProperties();
-        properties.put(ID, scd.getId());
-
-        this.metadata.createDataSource(toUse, dsType, encryptionComponent.decrypt(properties));
+        TeiidDataSource teiidDS = scd.createDataSource(toUse);
```
Comment (shawkins) [inline server/komodo-server/src/main/java/org/komodo/openshift/TeiidOpenShiftClient.java]: Maybe pass the DefaultSyndesisDataSource into a registerDataSource, then we can just get rid of the getDataSource check and handle everything in the DefaultMetadataInstance.

File: server/komodo-server/src/main/java/org/komodo/metadata/internal/DefaultMetadataInstance.java [hunk 2]
```diff
@@ -353,14 +370,24 @@ public TeiidDataSourceImpl getDataSource(String name) throws KException {
     }
 
     @Override
-    public void deleteDataSource(String dsName) throws KException {
-        TeiidDataSourceImpl ds = this.datasources.remove(dsName);
-        if (ds != null) {
-            this.server.removeConnectionFactoryProvider(dsName);
+    public synchronized void deleteDataSource(String dsName) throws KException {
```
[the remark this answers is not in the document]
Comment (rareddy) [reply server/komodo-server/src/main/java/org/komodo/metadata/internal/DefaultMetadataInstance.java]: corrected as suggested

File: server/komodo-server/src/main/java/org/komodo/openshift/TeiidOpenShiftClient.java (hunk 1 above)
Comment (rareddy) [reply server/komodo-server/src/main/java/org/komodo/openshift/TeiidOpenShiftClient.java]: corrected as suggested

**Comment by rareddy** [comment]
@shawkins I have updated the code per your suggestions.

**Comment by shawkins** [comment]
Merged this with additional changes to consolidate down to a single map of sources.  Also adding better handling of the openshift client in a follow-on commit to help with the integration test.

**Pull request merged**